custom_components/unifiprotect/unifi_protect_server.py

```diff
@@ -40,293 +53,248 @@ def __init__(
         self.device_data = {}
 
         urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
-        self.req = requests.session()
-        self._api_auth_bearer_token = self._get_api_auth_bearer_token()
-        self.update()
+        self.req = session
+        self._api_auth_bearer_token = None
 
     @property
     def devices(self):
         """ Returns a JSON formatted list of Devices. """
         return self.device_data
 
-    def update(self):
+    async def update(self) -> dict:
         """Updates the status of devices."""
-        self._get_camera_list()
-        self._get_motion_events(10)
+        if self._api_auth_bearer_token is None:
+            self._api_auth_bearer_token = await self._get_api_auth_bearer_token()
+        await self._get_camera_list()
+        await self._get_motion_events(10)
+        return self.devices
 
-    def _get_api_auth_bearer_token(self):
+    async def _get_api_auth_bearer_token(self) -> str:
         """get bearer token using username and password of local user."""
-
```

Comment:
I cleaned up a bunch of these strings to use `f` strings and use a common base_url instead of repeating it in all of the methods.